### PR TITLE
Feature/eui 1119 openid connect session

### PR DESCRIPTION
**Pull request by fazanki**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: api/auth/index.ts [hunk 1]
```diff
@@ -1,181 +1,157 @@
...
-              axios.defaults.headers.common.ServiceAuthorization = token
-            } else {
-                logger.warn('Cannot attach S2S token ')
-                doLogout(req, res, 401)
-            }
-        }
 
-        next()
+    // if client is already configured, exit early
+    if (app.locals.client) {
```
Comment (paddybasi) [inline api/auth/index.ts]: I've had to remove this fyi, for some reason it was still configuring the client with an IP address despite the test above

File: package.json [hunk 2]
```diff
@@ -11,9 +11,9 @@
     "start:node": "cd api && yarn watch",
     "start:node:docker": "PUI_ENV=ldocker;cd api && yarn watch",
     "test:node": "cd api && yarn test",
-    "test:smoke": "sh whitelist.sh && ts-node ./test/integration/tests/test.ts",
+    "test:smoke": "sh whitelist.sh",
     "test:smoke:functional": "sh whitelist.sh && webdriver-manager update --versions.chrome 2.40 && protractor ./test/e2e/config/smoke.conf.js --local",
-    "test:functional": "webdriver-manager update --versions.chrome 2.40 && protractor ./test/e2e/config/functional.conf.js",
+    "test:functional": "sh whitelist.sh",
```
Comment (paddybasi) [inline package.json]: we don't really need to whitelist twice

File: sonar-project.properties [hunk 3]
```diff
@@ -3,9 +3,8 @@ sonar.projectName=rpx-xui-mo-webapp
 sonar.projectVersion=0.0.1
 
 sonar.sources=src/app/
-sonar.exclusions= api/**, test/**, config/**, api/**/templates/**, src/public/**, src/environments/**, **/*.html, **/*.scss, src/app/app.constants.ts, src/app/utils/app-utils.ts, /app.constants.ts,, src/app/services/appInsightsWrapper.ts
+sonar.exclusions= api/**, test/**, config/**, api/**/templates/**, src/public/**, src/environments/**, **/*.html, **/*.scss, src/app/app.constants.ts, src/app/utils/app-utils.ts, /app.constants.ts, src/app/services/appInsightsWrapper.ts, *.guard.ts
```
Comment (paddybasi) [inline sonar-project.properties]: hmm, we should really be testing guards

File: api/user/index.ts [hunk 4]
```diff
@@ -2,17 +2,47 @@ import * as express from 'express'
...
+  }
+  console.log(req.session.passport.user)
   const UserDetails: UserProfileModel = {
-    email: req.session.auth.email,
-    orgId: req.session.auth.orgId,
-    roles: req.session.auth.roles,
-    userId: req.session.auth.userId
+    email: req.session.passport.user.userinfo.sub,
+    idleTime: getUserTimeouts(),
+    orgId: req.session.passport, // todo find org id
```
Comment (paddybasi) [inline api/user/index.ts]: this contains the refresh_token

File: package.json (hunk 2 above)
Comment (fazanki) [reply package.json]: vamshi says NO

**Pull request closed without merging**